### DISCUSSION TITLE
How to stop VCPKG from re-building every package randomly.

### DISCUSSION THREAD

**Discussion by Cazadorro** [microsoft/vcpkg#33842] [Q&A]
Every week VCPKG seems to want to rebuild every package for seemingly no reason.  I depend on OpenCV, which under the best conditions takes an hour to build.  My company's security settings however slows things down to the point it can take a whole day just to build OpenCV, this makes it very frustrating if not right out impossible to use VCPKG right now.  Is there a way I can just tell VCPKG to stop trying to re-build everything? I'm using Manifest mode and CMake integration, along with CMakePresets.json with my projects. 

This issue appears to be a bug from a year ago that has yet to be resolved : https://github.com/microsoft/vcpkg/issues/26601 But I'm not asking for a solution to that, I just want to stop rebuilding behavior as a bandaid entirely.

**Comment by autoantwort** [comment]
Use a fixed commit of vcpkg and then it don't rebuilds everything. Then it only rebuilds if your update your compiler.

**Reply by Cazadorro** [reply]
I haven't updated the vcpkg repo in months so that shouldn't be an issue right? Is there a way to figure out what triggered the rebuild?

**Reply by autoantwort** [reply]
You can check the `vcpkg_abi_info.txt` files and see how they differ

**Reply by Cazadorro** [reply]
what is the "they" in "how they differ".  I see there's a "vcpkg_abi_info.txt" per port, but I don't see how to use that to compare to anything.

**Reply by autoantwort** [reply]
You can compare that against your local binary cache that caches the previous versions of things your build. The default location is 
<img width="657" alt="image" src="https://github.com/microsoft/vcpkg/assets/41973254/b0baee64-649a-4247-bb12-8037900df0e8">

The downside is that you don't know in which zip which package is saved. You could extract random ones and compare the `vcpkg_abi_info.txt` file in there with the file in your installed folder. An maybe simpler approach is the use https://github.com/microsoft/vcpkg-tool/pull/325 and call `vcpkg binarycache list` (you have to build the vcpkg-tool yourself to use this command)

**Comment by Osyotr** [comment]
Try this: https://github.com/microsoft/vcpkg/issues/26346#issuecomment-1319244766
